src/middlewared/middlewared/plugins/idmap.py

```diff
@@ -1081,35 +1081,37 @@ async def builtins(self, filters, options):
         return filter_list(out, filters, options)
 
     @private
-    async def id_to_name(self, id_, id_type):
+    async def id_to_name(self, xid, id_type):
         idtype = IDType[id_type]
         idmap_timeout = 5.0
-
-        if idtype == IDType.GROUP or idtype == IDType.BOTH:
-            method = "group.get_group_obj"
-            to_check = {"gid": id_}
-            key = 'gr_name'
-        elif idtype == IDType.USER:
-            method = "user.get_user_obj"
-            to_check = {"uid": id_}
-            key = 'pw_name'
-        else:
-            raise CallError(f"Unsupported id_type: [{idtype.name}]")
+        options = {'extra': {'additional_information': ['DS']}, 'get': True}
+
+        match idtype:
+            case IDType.GROUP | IDType.BOTH:
```

Comment:
While not directly part of this PR, could you add some comments in `idmap_constants.py` with a simple explanation of what `BOTH` means?  I _assume_ it means 'either', but... 🤷‍♂️